tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+        metavar='CONTAINERFILE_WORKDIR',
+        help="""
+             Select working directory for the podman build invocation.
+             """)
+
+    containerimage: Optional[str] = field(
+        default=None,
+        option=('--containerimage'),
+        metavar='CONTAINERIMAGE',
+        help="""
+             Select container image to be used to build a bootc disk.
+             This takes priority over containerfile.
+             """)
+
+    add_deps: bool = field(
+        default=True,
+        is_flag=True,
+        option=('--add-deps'),
+        help="""
+             Add tmt dependencies to the supplied container image or image built
+             from the supplied Containerfile.
+             This will cause a derived image to be built from the supplied image.
+             """)
+
+    image_builder: str = field(
+        default=DEFAULT_IMAGE_BUILDER,
+        option=('--image-builder'),
+        metavar='IMAGEBUILDER',
+        help="""
+             The full repo:tag url of the bootc image builder image to use for
+             building the bootc disk image.
+             """)
+
+    def set_image(self, image: str) -> None:
+        self.image = image
+
+
+@tmt.steps.provides_method('bootc')
+class ProvisionBootc(tmt.steps.provision.ProvisionPlugin[BootcData]):
+    """
+    Provision a local virtual machine using a bootc container image
+
+    Minimal config which uses the Fedora bootc image:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            containerimage: quay.io/fedora/fedora-bootc:40
+
+    Here's a config example using a containerfile:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            containerfile: "./my-custom-image.containerfile"
+            containerfile-workdir: .
+            image_builder: quay.io/centos-bootc/bootc-image-builder:stream9
+            disk: 100
+
+    Another config example using an image that includes tmt dependencies:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            add_deps: false
+            containerimage: localhost/my-image-with-deps
+
+    This plugin is an extension of the virtual.testcloud plugin.
+    Essentially, it takes a container image as input, builds a
+    bootc disk image from the container image, then uses the virtual.testcloud
+    plugin to create a virtual machine using the bootc disk image.
+
+    The bootc disk creation requires running podman as root, this is typically
+    done by running the command in a rootful podman-machine. The podman-machine
+    also needs access to ``/var/tmp/tmt``. An example command to initialize the
+    machine:
+
+    .. code-block:: shell
+
+        podman machine init --rootful --disk-size 200 --memory 8192 \
+        --cpus 8 -v /var/tmp/tmt:/var/tmp/tmt -v $HOME:$HOME
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+    _id = str(uuid.uuid4())[:8]
```

Comment:
What is the purpose of this id? Would it be possible to use `Run.unique_id`, https://github.com/teemtee/tmt/blob/main/tmt/base.py#L3317? It should be available somewhere from `self.parent.step.my_run.unique_id`.

We should think twice before adding new kind ID, especially when it's with no clear connection to run owning images being built, might be much harder to locate them when removing things.

Reply:
Thanks, I wasn't sure how to access the unique run id so I just generated one. It is appended to the container images tags to keep them unique for cleanup, etc. I switched to using `self.parent.step.my_run.unique_id`.

Reply:
The attribute is no longer used, and it should be removed.